contributors.d

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env rdmd
+
+/**
+Query contributors between two D releases.
+
+Copyright: D Language Foundation 2017.
+
+License:   $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Example usage:
+
+---
+./contributors.d "v2.074.0..v2.075.0"
+---
+
+Author: Sebastian Wilzbach
+*/
+
+import std.array;
+import std.algorithm;
+import std.conv;
+import std.exception;
+import std.file;
+import std.format;
+import std.process;
+import std.path;
+import std.range;
+import std.stdio;
+import std.string;
+import std.typecons;
+
+/// Name <my@email.com>
+struct GitAuthor
+{
+    string name, email;
+}
+
+/// Options for finding authors
+struct FindConfig
+{
+    bool refreshTags; /// will query github.com for new tags
+    bool noMerges; // will ignore merge commits
+    bool showAllContributrs; // will ignore the revRange and show all contributors
+    string cwd; // working directory (should be tools)
+}
+
+/// Used to map multiple authors into one
+GitAuthor[GitAuthor] mappings;
+
+/**
+Search all git commit messages within revRange of all D repositories
+Returns: Array that maps each git `Author: ...` line to a GitAuthor
+*/
+auto findAuthors(string revRange, FindConfig config)
+{
+    Appender!(GitAuthor[]) authors;
+    int commits;
+    foreach (repo; ["dmd", "druntime", "phobos", "dlang.org", "tools", "installer"]
+             .map!(r => buildPath(config.cwd, "..", r)))
+    {
+        if (config.refreshTags)
+        {
+            auto cmd = ["git", "-C", repo, "fetch", "--tags", "https://github.com/dlang/" ~ repo.baseName,
+                               "+refs/heads/*:refs/remotes/upstream/*"];
+            auto p = pipeProcess(cmd, Redirect.stdout);
+            enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
+        }
+
+        auto cmd = ["git", "-C", repo, "log"];
+        if (!config.showAllContributrs)
+            cmd ~= revRange;
+        if (config.noMerges)
+            cmd ~= "--no-merges";
+
+        auto p = pipeProcess(cmd, Redirect.stdout);
+        scope(exit) enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
+
+        static immutable Author = "Author: ";
+        p.stdout
+            .byLineCopy
+            .filter!(line => line.startsWith(Author))
+            .tee!(_ => commits++)
+            .map!((line){
+                line.skipOver(Author);
+                return line.readUser;
+            })
+            .filter!(a => a.name != "The Dlang Bot")
```

Comment:
Maybe you'd rather want to filter out merges, [`--no-merges`](https://git-scm.com/docs/git-log).

Reply:
There's already a flag to change to this behavior. However, I wasn't sure whether we really want to exclude merges by default.